components/inputs/test/input-search.visual-diff.js

```diff
@@ -58,21 +57,13 @@ describe('d2l-input-search', () => {
 	});
 
 	it('focus-search-button', async function() {
-		await page.evaluateHandle(() => {
-			forceFocusVisible(
-				document.querySelector('#no-value').shadowRoot.querySelector('d2l-button-icon[icon="tier1:search"]')
-			);
-		});
+		await focusWithKeyboard(page, ['#no-value', 'd2l-button-icon[icon="tier1:search"]']);
```

Comment:
Ah btw - after you've gone and added this array selector functionality for shadow roots, I remembered that we already have a solution using the `pierce/` selector in Puppeteer:
https://github.com/BrightspaceUI/core/blob/main/components/table/test/table.visual-diff.js#L153

Oops!

Reply:
Oh neat -- I'll try that out and remove the other solution.

Reply:
Anyone else looking here - it turns out that `pierce/` doesn't "partially pierce" the shadow DOM, meaning you can't have different parts of the selector in different shadow DOMs - so it's less useful than I was thinking. [Official doc](https://pptr.dev/guides/query-selectors#pierce-selectors-pierce)